hydroflow/Cargo.toml

```diff
@@ -60,6 +60,10 @@ tokio-util = { version = "0.7.4", features = [ "net", "codec" ] }
 variadics = { path = "../variadics" }
 rustc-hash = "1.1.0"
 
+# Diagnostics feature does not work on wasm32.
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+hydroflow_lang = { path = "../hydroflow_lang", features = ["diagnostics"] }
```

Comment:
Shouldn't this be passed thru `hydroflow_macro` ?

Reply:
Arbitrarily chose hydroflow_lang since it was the only one not marked as optional. Can change to hydroflow_macro if you prefer that, I have no opinion.

Reply:
Hmm yeah, I think we should have the feature in `hydroflow_macro` as well and go thru that, only have diagnostics enabled if the `macros` feature is enabled

Reply:
Done, PTAL.

Reply:
Hmm even when targeting `wasm32` in hydroflow we should have diagnostics enabled, right? Because those diagnostics are for the macro which isn't being compiled to `wasm32` regardless...

Reply:
That's what I was thinking, if wasm is only the target we should still be able to have diagnostics

Reply:
Although, we don't even need these lines because when we build `hydroflow_datalog`, we will enable the diagnostic features anyway. So I think I can remove these lines from hydroflow/Cargo.toml entirely.

Reply:
When we do [the macro parsing] compilation [step] in browser e.g. for the demo, that is when diagnostics dont work